service/app/util/Filters.java

```diff
@@ -1,47 +1,42 @@
 package util;
 
-import javax.inject.Inject;
-
-import play.api.mvc.EssentialFilter;
-import play.http.HttpFilters;
-import play.filters.gzip.GzipFilter;
-import play.filters.gzip.GzipFilterConfig;
-import akka.stream.Materializer;
 import filters.LoggingFilter;
+import javax.inject.Inject;
+import org.apache.http.HttpHeaders;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.request.HeaderParam;
 import play.api.mvc.EssentialAction;
+import play.api.mvc.EssentialFilter;
 import play.api.mvc.RequestHeader;
 import play.api.mvc.ResponseHeader;
 import play.filters.gzip.Gzip;
 import play.filters.gzip.GzipFilter;
+import play.http.HttpFilters;
 import scala.Function2;
 import scala.runtime.AbstractFunction2;
-import org.apache.commons.lang3.StringUtils;
-import org.apache.http.HttpHeaders;
-import org.sunbird.common.models.util.JsonKey;
-import org.sunbird.common.models.util.LoggerEnum;
-import org.sunbird.common.models.util.ProjectLogger;
-import org.sunbird.common.models.util.ProjectUtil;
-import org.sunbird.common.models.util.PropertiesCache;
 
 public class Filters implements HttpFilters {
   private EssentialFilter[] filters;
-  private static boolean GzipFilterEnabled = Boolean
-      .parseBoolean(ProjectUtil.getConfigValue(JsonKey.SUNBIRD_GZIP_ENABLE));
+  private static boolean GzipFilterEnabled =
+      Boolean.parseBoolean(ProjectUtil.getConfigValue(JsonKey.SUNBIRD_GZIP_ENABLE));
   private static final String GZIP = "gzip";
   // Size of buffer to use for gzip.
   private static final int BUFFER_SIZE = 8192;
   // Content length threshold, after which the filter will switch to chunking the result.
-  private static final int  CHUNKED_THRESHOLD = 102400;
+  private static final int CHUNKED_THRESHOLD = 102400;
 
-  private Function2<RequestHeader, ResponseHeader, Object> shouldGzip = new AbstractFunction2<RequestHeader, ResponseHeader, Object>() {
+  public static final double sunbird_gzip_size_threshold = 256000.0;
```

Comment:
This value should be read from config similar to sunbird_gzip_enable

Reply:
following value set to config file ```export sunbird_gzip_size_threshold=262144```

Reply:
Also, add this in external resource properties file.